source/MaterialXGenHw/Nodes/HwSurfaceNode.cpp

```diff
@@ -89,10 +90,10 @@ void HwSurfaceNode::emitFunctionCall(const ShaderNode& node, GenContext& context
 
         shadergen.emitScopeBegin(stage);
 
-        shadergen.emitLine("vec3 N = normalize(" + prefix + HW::T_NORMAL_WORLD + ")", stage);
-        shadergen.emitLine("vec3 V = normalize(" + HW::T_VIEW_POSITION + " - " + prefix + HW::T_POSITION_WORLD + ")", stage);
-        shadergen.emitLine("vec3 P = " + prefix + HW::T_POSITION_WORLD, stage);
-        shadergen.emitLine("vec3 L = vec3(0,0,0);", stage);
+        shadergen.emitLine(syntax.getTypeName(Type::VECTOR3)+" N = normalize(" + prefix + HW::T_NORMAL_WORLD + ")", stage);
+        shadergen.emitLine(syntax.getTypeName(Type::VECTOR3)+" V = normalize(" + HW::T_VIEW_POSITION + " - " + prefix + HW::T_POSITION_WORLD + ")", stage);
+        shadergen.emitLine(syntax.getTypeName(Type::VECTOR3)+" P = " + prefix + HW::T_POSITION_WORLD, stage);
+        shadergen.emitLine(syntax.getTypeName(Type::VECTOR3)+" L = "+syntax.getValue(Type::VECTOR3, HW::zeroVec3), stage);
```

Comment:
Should we trust the compiler to optimize repeated function calls, or perhaps proactively remove the redundant calls ourselves? This pattern applies to a few other places in this PR as well.